### PR TITLE
Handle datetimes and empty selections better for inspect operations

### PR DESCRIPTION
**Pull request by philippjfr**
A number of fixes for `inspect` operations:

- Initialize with empty x/y coordinate
- Handle undefined x/y coordinates
- Allow datetime axis
- Allow axes with different dimensions, e.g. datetime and numeric (by normalizing space before computing distances)

### CODE REVIEW

**Review by jbednar — COMMENTED** [review]
Thanks! Those changes look good to me, though I'm not an expert in datetime dtypes.

File: holoviews/operation/datashader.py [hunk 1]
```diff
@@ -1990,11 +2003,24 @@ def _sort_by_distance(cls, raster, df, x, y):
...
+        # coordinate space to ensure that distance
+        # in both direction is handled the same.
+        if xtype != ytype and len(dx) and len(dy):
+            with warnings.catch_warnings():
+                warnings.filterwarnings('ignore', r'invalid value encountered in (divide)')
+                dx = (dx - dx.min()) / (dx.max() - dx.min())
+                dy = (dy - dy.min()) / (dy.max() - dy.min())
+        distances = pd.Series(dx**2 + dy**2)
+        distances[distances.isna()] = 0
         return df.iloc[distances.argsort().values]
```
Comment (hoxbro) [inline holoviews/operation/datashader.py:2021]:
```suggestion
        return df.iloc[distances.fillna(0).argsort().values]
```

File: holoviews/operation/datashader.py [hunk 2]
```diff
@@ -1990,11 +2003,24 @@ def _sort_by_distance(cls, raster, df, x, y):
...
+            dy = dx.astype('int64')
+        # If coordinate types don't match normalize
+        # coordinate space to ensure that distance
+        # in both direction is handled the same.
+        if xtype != ytype and len(dx) and len(dy):
+            with warnings.catch_warnings():
+                warnings.filterwarnings('ignore', r'invalid value encountered in (divide)')
+                dx = (dx - dx.min()) / (dx.max() - dx.min())
+                dy = (dy - dy.min()) / (dy.max() - dy.min())
+        distances = pd.Series(dx**2 + dy**2)
```
Comment (hoxbro) [inline holoviews/operation/datashader.py:2019]: Maybe use pd.array here?

**Review by hoxbro — APPROVED** [review]
Some nit comments.

File: holoviews/operation/datashader.py [hunk 3]
```diff
@@ -1990,9 +2003,20 @@ def _sort_by_distance(cls, raster, df, x, y):
...
+            dx = dx.astype('int64')
+        if ytype in 'Mm':
+            dy = dx.astype('int64')
+        # If coordinate types don't match normalize
+        # coordinate space to ensure that distance
+        # in both direction is handled the same.
+        if xtype != ytype and len(dx) and len(dy):
+            with np.errstate(divide='ignore'):
+                dx = (dx - dx.min()) / (dx.max() - dx.min())
+                dy = (dy - dy.min()) / (dy.max() - dy.min())
```
Comment (philippjfr) [inline holoviews/operation/datashader.py:2017]:
```suggestion
            with warnings.catch_warnings():
                warnings.filterwarnings('ignore', r'invalid value encountered in (divide)')
                dx = (dx - dx.min()) / (dx.max() - dx.min())
                dy = (dy - dy.min()) / (dy.max() - dy.min())
```

**Pull request merged**